### PR TITLE
Develop anton

### PR DESCRIPTION
**Pull request by antonkorotkov**
Includes issues:

- #275 
- #430

### CODE REVIEW

File: lib/classes/compatibility/divi.php [hunk 1]
```diff
@@ -29,7 +52,25 @@ public function module_init($sm){
                 ) {
                     remove_filter( 'sanitize_file_name', array( "wpCloud\StatelessMedia\Utility", 'randomize_filename' ), 10 );
                 }
-                
+
+                // maybe disable the filter
+                add_filter('sanitize_file_name', array( $this, 'sanitize_file_name' ), 1);
```
Comment (alimuzzaman) [inline lib/classes/compatibility/divi.php]: You could have used `stateless_skip_cache_busting` filter instead. Returning true will short-circuit the [randomize_filename](https://github.com/wpCloud/wp-stateless/blob/v2.3/lib/classes/class-utility.php#L78) function.

File: lib/classes/class-utility.php [hunk 2]
```diff
@@ -602,8 +602,55 @@ function_exists('is_wp_version_compatible') &&
...
+       *  array(
+       *    'stack_level' => 4,
+       *    'function' => 'get_cache_filename',
+       *    'class' => 'ET_Builder_Element'
+       *  )
+       * )
+       *
+       * @return bool
+       */
+      public static function isCallStackMatches( $callstack, $conditions ) {
```
Comment (alimuzzaman) [inline lib/classes/class-utility.php:628]: Cool

File: lib/classes/compatibility/divi.php (hunk 1 above)
Comment (antonkorotkov) [reply lib/classes/compatibility/divi.php]: Ok, I'll check that filter. But don't understand now what exactly will short-circuit. Can you give an example?

Comment (alimuzzaman) [reply lib/classes/compatibility/divi.php]: Check here https://github.com/wpCloud/wp-stateless/blob/v2.3/lib/classes/class-utility.php#L78
Here is an example https://github.com/wpCloud/wp-stateless/blob/v2.3/lib/classes/compatibility/buddyboss.php

Comment (antonkorotkov) [reply lib/classes/compatibility/divi.php]: Ah, I see now what you mean short-circuit. Will implement using that filter, thanks.

Comment (antonkorotkov) [reply lib/classes/compatibility/divi.php]: @alimuzzaman I think there is some bug in this line https://github.com/wpCloud/wp-stateless/blob/a93638abe0d2ae5013e0c7f6f7155e8e101c0c33/lib/classes/class-utility.php#L80

It should return `$filename` but not `$return`. If you return `$return` then the result of the `randomize_filename` function will be `1` (casted from true)

Can you confirm this?

Comment (alimuzzaman) [reply lib/classes/compatibility/divi.php]: Sorry, for the confusion. you should return unmodified $filename from the filter, not true. Then $return will become $filename.
I wrote like that so you can modify it from the filter if needed.

Comment (antonkorotkov) [reply lib/classes/compatibility/divi.php]: > Sorry, for the confusion. you should return unmodified $filename from the filter, not true. Then $return will become $filename.
> I wrote like that so you can modify it from the filter.

Ok, makes sense. Better to keep it as is.

**Review by alimuzzaman — APPROVED** [review]

File: lib/classes/compatibility/divi.php (hunk 1 above)
Comment (alimuzzaman) [reply lib/classes/compatibility/divi.php]: I will add documentation in `stateless_skip_cache_busting` filter.

**Review by alimuzzaman — APPROVED** [review]

**Pull request merged**